os/drivers/heapinfo/heapinfo_drv.c

```diff
@@ -88,14 +88,14 @@ static int heapinfo_ioctl(FAR struct file *filep, int cmd, unsigned long arg)
 		switch (option->heap_type) {
 #ifndef CONFIG_BUILD_PROTECTED
 		case HEAPINFO_HEAP_TYPE_KERNEL:
-			heap = &BASE_HEAP[regionx_heap_idx[0]];
+			heap = &USR_HEAP[regionx_heap_idx[0]];
```

Comment:
Because of naming, this looks strange. This means getting the kernel heap from user heap data.

Reply:
I understand that the name seems odd, but in flat build, all kernel memory operations get redirected to user mm and internally, USR_HEAP only is getting used. However, I could add a comment here to indicate the reason for using USR_HEAP for TYPE_KERNEL. Is that ok?

Reply:
ok.

Reply:
Done